ambari-server/src/main/python/ambari_server/setupSecurity.py

```diff
@@ -716,14 +716,16 @@ def setup_ldap(options):
     err = 'Ambari Server is not running.'
     raise FatalException(1, err)
 
-  current_client_security = get_value_from_properties(properties,CLIENT_SECURITY,"no auth method")
-  if current_client_security != 'ldap':
-    query = "Currently '" + current_client_security + "' is configured, do you wish to use LDAP instead [y/n] (n)? "
-    if get_YN_input(query, False):
-      pass
-    else:
-      err = "Currently '" + current_client_security + "' configured. Can not setup LDAP."
-      raise FatalException(1, err)
+  enforce_ldap = True if options.ldap_enforcement is not None and options.ldap_enforcement == 'true' else False
+  if not enforce_ldap:
+    current_client_security = get_value_from_properties(properties,CLIENT_SECURITY,"no auth method")
+    if current_client_security != 'ldap':
+      query = "Currently '" + current_client_security + "' is configured, do you wish to use LDAP instead [y/n] (n)? "
+      if get_YN_input(query, False):
```

Comment:
Instead of introducing a new flag to force LDAP setup, you might consider defaulting this input to `True`.  This would allow `setup-ldap --silent` to successfully complete, while still giving users a chance to quit at this point when invoked without `--silent`.  If this is too drastic to apply in all cases, then it still can be considered for the `no auth configured` case.

Reply:
It's been defaulted to False for a reason; let me discuss this with @rlevas first.

Reply:
Implemented that for the case you suggested (when no auth. method is configured in the first place)